### PR TITLE
add chaincode generation endpoint with integration test

### PR DESCRIPTION
**Pull request by pallabez**

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -265,4 +265,6 @@ router.get('/:username', users.getUser)
 // upload.single('profile') -> multer inmemory storage of file for type multipart/form-data
 router.post('/picture', authenticate, upload.single('profile'), users.postUserPicture)
 
+router.patch('/chainCode', authenticate, users.updateChainCode)
```
Comment (swarajpure) [inline routes/users.js:268]: What's the reason of going ahead with patch request here?

Comment (pallabez) [reply routes/users.js:268]: We are updating the chain code. There might already be an existing chain code in the user object so we can't use POST here.

Comment (pallabez) [reply routes/users.js:268]: The endpoint is used for generating a new chain code for that user, regardless of if the person have a pre existing chain code or not.

File: controllers/users.js [hunk 2]
```diff
@@ -143,11 +144,33 @@ const postUserPicture = async (req, res) => {
   }
 }
 
+/**
+ * Post user profile picture
```
Comment (harshith-venkatesh) [inline controllers/users.js:148]: Please add comment for `updateChainCode`

File: routes/users.js (hunk 1 above)
Comment (harshith-venkatesh) [reply routes/users.js:268]: `The endpoint is used for generating a new chain code for that user` this is bit contradicting the above statement here

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: controllers/users.js (hunk 2 above)
Comment (pallabez) [reply controllers/users.js:148]: Right. Thanks for pointing it out.

File: controllers/users.js [hunk 3]
```diff
@@ -145,19 +145,19 @@ const postUserPicture = async (req, res) => {
...
  * @param req {Object} - Express request object
  * @param res {Object} - Express response object
  */
-const updateChainCode = async (req, res) => {
+const updateChaincode = async (req, res) => {
   try {
     const key = uuid.v4()
     const { id: userId } = req.userData
-    await userQuery.addOrUpdate({ chainCode: key }, userId)
+    await userQuery.addOrUpdate({ chaincode: key }, userId)
```
Comment (lakshayman) [inline controllers/users.js:157]: @pallabez can you remove this line as it is not needed
We discussed in last call

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: controllers/users.js (hunk 3 above)
Comment (pallabez) [reply controllers/users.js:157]: got it. 👍

**Pull request closed without merging**